src/compiler/checker.ts

```diff
@@ -8823,6 +8932,115 @@ module ts {
             }
         }
 
+        function getPromisedType(type: Type): Type {
+            // the "promised type" of a type is the type of the "value" argument of the "onfulfilled" callback.

```

Comment:
Promote this to a JSDoc comment, use singe quotes around "value" and "onfulfilled"